### PR TITLE
Add option to choose from registered hatches to toolbar

### PR DESCRIPTION
**Pull request by jpirnay**
image

image

### CODE REVIEW

File: meerk40t/core/elements/shapes.py [hunk 1]
```diff
@@ -234,17 +234,50 @@ def element_line(command, x0, y0, x1, y1, data=None, post=None, **kwargs):
         post.append(classify_new(data))
         return "elements", data
 
+
```
Comment (tatarize) [inline meerk40t/core/elements/shapes.py:237]: I'm not sure shapes is entirely the right place for these commands. Not sure a better place but we might need to sort these properly at some point.

**Comment by tiger12506** [comment]
I took a look at this and it seems harmless. Works smoothly.

**Review by tiger12506 — APPROVED** [review]
I took a look at this. Looks harmless, works smoothly.

File: meerk40t/gui/wxmmain.py [hunk 2]
```diff
@@ -1026,9 +1030,75 @@ def open_property_window_for_node(self, node):
 
     @staticmethod
     def sub_register(kernel):
+        def register_effects():
```
Comment (tatarize) [inline meerk40t/gui/wxmmain.py:1033]: Not really sure wxmain is the right place for these either.

File: meerk40t/gui/wxmmain.py [hunk 3]
```diff
@@ -1130,19 +1200,103 @@ def contains_an_element():
...
+            "tip":  sub_effects[0]["tip"],
+            "action":  sub_effects[0]["action"],
+            "action_right": lambda v: kernel.elements("effect-remove\n"),
+            "size": bsize_normal,
+            "rule_enabled": lambda cond: contains_an_element(),
+        }
+
+        if len(sub_effects) > 1:
+            hatch_button["multi"] = sub_effects
+        # "multi": [
```
Comment (tatarize) [inline meerk40t/gui/wxmmain.py:1254]: This just random comment duplication of stuff elsewhere is probably unneeded.

**Review by tatarize — APPROVED** [review]
All in all good and will work.

**Pull request merged**